store-implementation/accumulo-store/src/test/java/uk/gov/gchq/gaffer/accumulostore/operation/handler/GetElementsWithinSetHandlerTest.java

```diff
@@ -165,30 +173,42 @@ public void shouldReturnElementsNoSummarisationGaffer1Store() throws OperationEx
     private void shouldReturnElementsNoSummarisation(final AccumuloStore store) throws OperationException {
         final GetElementsWithinSet operation = new GetElementsWithinSet.Builder().view(defaultView).input(seeds).build();
         final GetElementsWithinSetHandler handler = new GetElementsWithinSetHandler();
-        final CloseableIterable<? extends Element> elements = handler.doOperation(operation, user, store);
-
-        //Without query compaction the result size should be 5
-        final Set<Element> elementSet = Sets.newHashSet(elements);
-        assertThat(elementSet).hasSize(5);
-        assertEquals(Sets.newHashSet(expectedEdge1, expectedEdge2, expectedEdge3, expectedEntity1, expectedEntity2), elementSet);
-        for (final Element element : elementSet) {
-            if (element instanceof Edge) {
-                assertEquals(EdgeId.MatchedVertex.SOURCE, ((Edge) element).getMatchedVertex());
+        Iterable<? extends Element> elements = null;
+        try {
+            elements = handler.doOperation(operation, user, store);
+            // Without query compaction the result size should be 5
+            final Set<Element> elementSet = Sets.newHashSet(elements);
+            assertThat(elementSet).hasSize(5);
+            assertThat(elementSet).containsOnly(expectedEdge1, expectedEdge2, expectedEdge3, expectedEntity1, expectedEntity2);
+            for (final Element element : elementSet) {
+                if (element instanceof Edge) {
+                    assertThat(((Edge) element).getMatchedVertex()).isEqualTo(EdgeId.MatchedVertex.SOURCE);
+                }
             }
+        } finally {
```

Comment:
Think it was flagged by eclipse  so i added. Will check.

Reply:
I think I was thinking about #2583 when doing this. The `CloseableIterable` will become `Iterable` so you would need to do the `CloseableUtil` check. I can remove it...

Reply:
If this future proofs it from #2583 then I think it is fine. Will all of the tests need to change like this?

Reply:
As in, will they needs try - finally as part of #2583?

Reply:
Possibly, I think I may have done the necessary changes as part of the PR. I'll check

Reply:
It appears I have missed the check. I've added a note.

Reply:
So as I understand it, every time there is something like: `elements = something.doOperation`, where an iterable of elements is returned, there must now be a try - finally block to close the iterable. This is in case something breaks and the iterable isn't closed? Or it just always needs to be manually closed now? Did they close themselves before? I am just a bit confused, sorry.
Perhaps it can be removed from this PR and then added along with all the others in #2583? Then it will hopefully make sense to me in context.

Reply:
Yeah, had a discussion on how best to deal with having an `CloseableIterable` in Koryphe and Gaffer and decided on removing altogether and having the try-finally block. I'll remove the try-finally blocks.